### ISSUE TITLE
errorbar incorrectly accepts and plots Nx2 shaped yerr

### ISSUE THREAD

**Issue by wackywendell** [matplotlib/matplotlib#4984]
The docs for `errorbar` state:

> _xerr_/_yerr_: [ scalar | N, Nx1, or 2xN array-like ] If a scalar number, len(N) array-like object, or an Nx1 array-like object, errorbars are drawn at +/-value relative to the data.
> 
> If a sequence of shape 2xN, errorbars are drawn at -row1 and +row2 relative to the data.

However, if you pass a N×2 shaped array, the data still plots - but incorrectly. 

Here's an example:

``` python
x = np.arange(5) + .5
y = np.array([1,2,1.5, 2.5, 2.])
yerr = np.array([
        (.1,.2),
        (.1,.2),
        (.1,.2),
        (.1,.2),
        (.1,.2)
    ])

# correct behavior
plt.figure()
err = yerr.T
print(np.shape(err)) # (2, 5)
plt.errorbar(x, y, err)
plt.xlim(0, 5)

# incorrect behavior
plt.figure()
err = yerr
print(np.shape(err))
plt.errorbar(x, y, err) (5, 2)
plt.xlim(0, 5)
```

And here are the two plots:

![errorbar-correct](https://cloud.githubusercontent.com/assets/245071/9447356/eaa50976-4a64-11e5-90a1-488bbf136d56.png)
![errorbar-incorrect](https://cloud.githubusercontent.com/assets/245071/9447355/ea986ebe-4a64-11e5-89f0-91be1ec9d426.png)

**Comment by wackywendell** [comment]
I was looking at [the code](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/axes/_axes.py#L2821) for this, and it says things like "using list comps rather than arrays to preserve units". What exactly does that mean? What units? Wouldn't it be a lot cleaner if we were using numpy arrays here?

It looks like this changed [back in this commit](https://github.com/matplotlib/matplotlib/commit/39a01ad4f89f9632473430873a417f6ff558df5a#diff-ef7b5a7a3d15f6972a4644538157cae6L3702), all the way back in 2007, with just the commit message "fixed some examples bugs."

**Comment by WeatherGod** [comment]
matplotlib tries to support specialized data such as dates and other custom
things that maintain their own units information. This helps out greatly in
some other areas such as tick formatting. We recently had someone "fix" a
bunch of our functions to better support unitized data, but it looks like
he broke a couple of things that aren't in our unit tests. I'll see what I
can do to fix this up.

**Comment by WeatherGod** [comment]
Huh, I can't reproduce the problem. Just tried it out on master. It
correctly errors out for me:
```
>>> import matplotlib.pyplot as plt
>>> import numpy as np
>>> x = np.arange(5) + .5
>>> y = np.array([1.2, 1.5, 2.5, 2.])
>>> yerr = np.array([(.1, .2)]*5)
>>> plt.figure()
<matplotlib.figure.Figure object at 0x3349410>
>>> plt.errorbar(x, y, yerr)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File
"/nas/home/broot/centos6/lib/python2.7/site-packages/matplotlib-1.5.dev1-py2.7-linux-x86_64.egg/matplotlib/pyplot.py",
line 2823, in errorbar
    errorevery=errorevery, capthick=capthick, **kwargs)
  File
"/nas/home/broot/centos6/lib/python2.7/site-packages/matplotlib-1.5.dev1-py2.7-linux-x86_64.egg/matplotlib/axes/_axes.py",
line 2887, in errorbar
    in cbook.safezip(y, yerr)]
  File
"/nas/home/broot/centos6/lib/python2.7/site-packages/matplotlib-1.5.dev1-py2.7-linux-x86_64.egg/matplotlib/cbook.py",
line 1497, in safezip
    raise ValueError(_safezip_msg % (Nx, i + 1, len(arg)))
ValueError: In safezip, len(args[0])=4 but len(args[1])=5
```

**Comment by wackywendell** [comment]
Your code example has ***4*** points for `y`, and ***5*** for `yerr`. That's a different problem, that is correctly addressed... but not when `y` is length 5 and `yerr` is 5×2.

**Comment by WeatherGod** [comment]
Whoops, you are right, I misread a comma for a period.

**Comment by anntzer** [comment]
Closed by #4985.
